pkg/ottl/ottlfuncs/func_replace_all_patterns_test.go

```diff
@@ -241,7 +241,7 @@ func Test_replaceAllPatterns(t *testing.T) {
 			want: func(expectedMap pcommon.Map) {
 				expectedMap.PutStr("test", "hello world")
 				expectedMap.PutStr("test2", "hello")
-				expectedMap.PutStr("test3", "goodbye hash(world1) and hash(world2)")
+				expectedMap.PutStr("test3", "goodbye hash(world1) and world2")
```

Comment:
regex - (world1) and (world2) has two capture groups. We only hash the first - $1.

Reply:
the original string was `goodbye world1 and world2`. So with the above regex and replacement ($1) we only hash the first capture group and leave the rest of the string as is, resulting in

`goodbye hash(world1) and world2`

Reply:
I see what you mean, but in my opinion we should change that behavior. Right now without the optional function, `replace_all_patterns` will turn `goodbye world1 and world2` with replacement `$1` into `goodbye world1`. We should be calling the optional function on the entire replacement string (`$1`) and replacing the entire matched pattern with that string. So with the optional function `goodbye world1` becomes `goodbye hash(world1)`.

Reply:
ok this should be addressed in the latest [commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29408/commits/d80955615abddfd95e3512fb35b4aa8667457d90)